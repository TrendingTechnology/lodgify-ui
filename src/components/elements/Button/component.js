import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { isString } from 'lodash';
import { Button, Icon } from 'semantic-ui-react';

/**
 * A button indicates a possible user action.
 * @return {Object}
 */
export const Component = ({
  children,
  isDisabled,
  isLoading,
  isPositionedRight,
  isRounded,
  isSecondary,
  hasShadow,
  icon,
  size,
}) => (
  <Button
    disabled={isDisabled}
    floated={isPositionedRight ? 'right' : 'left'}
    loading={isLoading}
    secondary={isSecondary}
    circular={isRounded}
    size={size}
    className={cx({
      'has-icon': !!icon,
      'has-shadow': !!hasShadow,
    })}
  >
    {!icon ? null : (
      <div className="icon">
        {React.isValidElement(icon) ? <div className="icon">{icon}</div> : null}
        {isString(icon) ? <Icon name={icon} /> : null}
      </div>
    )}
    {children}
  </Button>
);

Component.displayName = 'Button';

Component.defaultProps = {
  isDisabled: false,
  isPositionedRight: false,
  isLoading: false,
  isRounded: false,
  isSecondary: false,
  hasShadow: false,
  icon: null,
  size: null,
};

Component.propTypes = {
  /** The text to display on the button. */
  children: PropTypes.node.isRequired,
  /** Is the button disabled. */
  isDisabled: PropTypes.bool,
  /** Is the button in loading state. */
  isLoading: PropTypes.bool,
  /** Is the button positioned on the right hand side of its container. */
  isPositionedRight: PropTypes.bool,
  /** Has the button rounded borders. */
  isRounded: PropTypes.bool,
  /** Is the button secondary. */
  isSecondary: PropTypes.bool,
  /** Has the button shadow. */
  hasShadow: PropTypes.bool,
  /** An icon to display in the button */
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /** The size of the button */
  size: PropTypes.string,
};
