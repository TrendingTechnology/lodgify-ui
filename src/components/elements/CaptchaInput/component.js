import React from 'react';
import PropTypes from 'prop-types';
import { Form, Image } from 'semantic-ui-react';

import { TextInput } from 'elements/TextInput';

/**
 * A captcha input helps to ensure a user is a real human.
 * @return {Object}
 */
export const Component = ({ image, label, name, onChange }) => (
  <Form.Group widths="equal">
    <Form.Field>
      <Image src={image} />
    </Form.Field>
    <TextInput label={label} name={name} onChange={onChange} />
  </Form.Group>
);

Component.displayName = 'CaptchaInput';

Component.defaultProps = {
  label: '',
  name: '',
  onChange: Function.prototype,
};

Component.propTypes = {
  /** The source url for the image to display. */
  image: PropTypes.string.isRequired,
  /** The visible label for the text input. */
  label: PropTypes.string,
  /** The name for the text input. */
  name: PropTypes.string,
  /**
   * A function called when the text input value changes
   * @param {String} name
   * @param {String} value
   */
  onChange: PropTypes.func,
};
