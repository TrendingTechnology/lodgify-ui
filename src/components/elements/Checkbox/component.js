import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'semantic-ui-react';

/**
 * A checkbox allows a user to select a value from a small set of options, often binary.
 * @return {Object}
 */
export const Component = ({
  isDisabled,
  isChecked,
  isCheckedByDefault,
  isRadioButton,
  isToggle,
  label,
  name,
  onChange,
  value,
}) => (
  <Checkbox
    checked={isChecked}
    defaultChecked={isCheckedByDefault}
    disabled={isDisabled}
    label={label}
    name={name}
    onChange={(event, { checked }) => onChange(name, checked)}
    radio={isRadioButton}
    value={value}
    toggle={isToggle}
  />
);

Component.displayName = 'Checkbox';

Component.defaultProps = {
  // Notes: Use undefined for uncontrolled checkboxes, boolean for controlled ones
  isChecked: undefined,
  isCheckedByDefault: undefined,
  isDisabled: false,
  isToggle: false,
  isRadioButton: undefined,
  label: '',
  name: undefined,
  onChange: Function.prototype,
  value: undefined,
};

Component.propTypes = {
  /** Is the checkbox checked. */
  isChecked: PropTypes.bool,
  /**
   * Is the checkbox initially checked.
   * This property is used only for uncontrolled checkboxes (when isChecked is not specified).
   * Refer to: https://reactjs.org/docs/uncontrolled-components.html
   * */
  isCheckedByDefault: PropTypes.bool,
  /** Is the checkbox disabled. */
  isDisabled: PropTypes.bool,
  /** Is the checkbox formatted as a radio button. Prefer using the <RadioButton> component instead */
  isRadioButton: PropTypes.bool,
  /** The label for the checkbox */
  label: PropTypes.string,
  /** Format to show an on or off choice. */
  isToggle: PropTypes.bool,
  /** The HTML input name. */
  name: PropTypes.string,
  /**
   * Event called when the user attempts to change the checked state.
   * @param {SyntheticEvent} event - React's original SyntheticEvent.
   * @param {object} data - All props and proposed checked/indeterminate state.
   */
  onChange: PropTypes.func,
  /** The HTML input value. */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};
