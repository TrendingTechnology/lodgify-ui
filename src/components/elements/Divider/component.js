import React from 'react';
import PropTypes from 'prop-types';
import { Divider } from 'semantic-ui-react';

/**
 * A divider adds whitespace between other elements.
 * @return {Object}
 */
export const Component = ({ hasLine }) => <Divider hidden={!hasLine} />;

Component.displayName = 'Divider';

Component.defaultProps = {
  hasLine: false,
};

Component.propTypes = {
  /** Does the divider have a visible line. */
  hasLine: PropTypes.bool,
};
