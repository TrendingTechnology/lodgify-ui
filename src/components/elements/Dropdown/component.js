import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Dropdown, Icon } from 'semantic-ui-react';
import getClassNames from 'classnames';

import { adaptOptions } from './utils/adaptOptions';
import { getDefaultValue } from './utils/getDefaultValue';

/**
 * A dropdown allows a user to select a value from a series of options.
 * @extends React.PureComponent
 */
export class Component extends PureComponent {
  state = {
    value: '',
  };

  // eslint-disable-next-line valid-jsdoc
  /**
   * Call the onChange function passed down via props
   * with the new state value
   */
  componentDidUpdate(prevProps, { value: prevValue }) {
    const { value } = this.state;
    const { name, onChange } = this.props;
    prevValue !== value && onChange(name, value);
  }

  // eslint-disable-next-line valid-jsdoc
  /**
   * Persist the value in component state
   */
  handleChange = (event, data) => {
    this.setState({ value: data.value });
  };

  render() {
    const { value } = this.state;
    const { label, options, icon } = this.props;
    const optionsWithImages = adaptOptions(options);
    const defaultValue = getDefaultValue(optionsWithImages);
    return (
      <div
        className={getClassNames('dropdown-container', {
          'has-images': optionsWithImages,
          'has-left-icon': icon,
        })}
      >
        <Dropdown
          className={getClassNames({
            dirty: value,
          })}
          className={value ? 'dirty' : ''}
          defaultValue={defaultValue}
          onChange={this.handleChange}
          options={optionsWithImages || options}
          placeholder={label}
          selection
        />
        {icon && <Icon name={icon} className="left-dropdown-icon" />}
      </div>
    );
  }
}

Component.displayName = 'Dropdown';

Component.defaultProps = {
  label: '',
  name: '',
  onChange: Function.prototype,
  icon: null,
};

Component.propTypes = {
  /** Icon for the dropdown */
  icon: PropTypes.string,
  /** The label for the dropdown. */
  label: PropTypes.string,
  /** The name for the dropdown. */
  name: PropTypes.string,
  /** A function called when the dropdown value changes. */
  onChange: PropTypes.func,
  /** The options which the user can select. */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      /** The source url for the image to display with the option. */
      image: PropTypes.string,
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.any,
    })
  ).isRequired,
};
