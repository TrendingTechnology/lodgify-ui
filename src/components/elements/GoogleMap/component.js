import React from 'react';
// import PropTypes from 'prop-types';
// import { Map } from 'semantic-ui-react';

/**
 * A map displays a location.
 * @return {Object}
 */
export const Component = () => <div>I'm totally a map</div>;

Component.displayName = 'GoogleMap';

// Component.defaultProps = {
//   hasLine: false,
// };
//
// Component.propTypes = {
//   /** Does the divider have a visible line. */
//   hasLine: PropTypes.bool,
// };
