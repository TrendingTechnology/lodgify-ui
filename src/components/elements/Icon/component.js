import React from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';

import { Paragraph } from 'typography/Paragraph';

import { getPath } from './utils/getPath';

/**
 * An icon is a glyph used to represent something else.
 * @return {Object}
 */
export const Component = ({
  color,
  isCircular,
  isColorInverted,
  isDisabled,
  label,
  name,
  path,
  size,
  ...otherProps
}) => (
  <i
    className={getClassNames('icon', color, size, {
      circular: isCircular,
      'inverted grey': isDisabled,
      inverted: isColorInverted,
    })}
    // Passing through props is required for
    // compatibility with dynamic components e.g. `ToolTip`
    {...otherProps}
  >
    <svg viewBox="0 0 24 24">
      <path d={getPath(name, path)} fill="currentColor" />
    </svg>
    {!!label && <Paragraph>{label}</Paragraph>}
  </i>
);

Component.displayName = 'Icon';

Component.defaultProps = {
  color: null,
  isCircular: false,
  isColorInverted: false,
  isDisabled: false,
  label: null,
  name: null,
  path: null,
  size: null,
};

Component.propTypes = {
  /** The color of the icon. */
  color: PropTypes.oneOf([
    'red',
    'orange',
    'yellow',
    'olive',
    'green',
    'teal',
    'blue',
    'violet',
    'purple',
    'pink',
    'brown',
    'grey',
    'black',
  ]),
  /** Is the icon formatted to appear circular  */
  isCircular: PropTypes.bool,
  /** Is the color of the icon inverted for contrast */
  isColorInverted: PropTypes.bool,
  /** Is the icon disabled. */
  isDisabled: PropTypes.bool,
  /** A visible label to display with the icon. */
  label: PropTypes.string,
  /** The name of the icon to display. Takes priority over `props.path`. */
  name: PropTypes.oneOf([
    'arrow down',
    'arrow left',
    'arrow right',
    'arrow up',
    'bathroom',
    'bedroom door',
    'bus',
    'calendar',
    'caret down',
    'caret left',
    'caret right',
    'caret up',
    'check in',
    'check out',
    'checkmark',
    'chevron left',
    'chevron right',
    'clock',
    'close',
    'coffee',
    'double bed',
    'fire',
    'guests',
    'home',
    'info',
    'leaf',
    'location',
    'map pin',
    'paw',
    'phone',
    'placeholder',
    'plane',
    'question mark',
    'road',
    'search',
    'single bed',
    'star',
    'sun',
    'train',
    'users',
    'wheelchair',
  ]),
  /**
   * A string containing a series of path descriptions. Should fit a 24x24 viewBox.
   * [See MDN for documentation.](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d)
   */
  path: PropTypes.string,
  /** The size of the icon. */
  size: PropTypes.oneOf([
    'mini',
    'tiny',
    'small',
    'large',
    'big',
    'huge',
    'massive',
  ]),
};
