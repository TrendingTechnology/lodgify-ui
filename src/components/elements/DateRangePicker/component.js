import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { isEqual, uniqueId } from 'lodash';
import { DateRangePicker } from 'react-dates';
import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';

import { Icon } from 'elements/Icon';

import { InputController } from '../InputController';

import { pickDatesFromState } from './pickDatesFromState';

/**
 * A date range picker lets a user pick a date range.
 * @extends {React.PureComponent}
 */
export class Component extends PureComponent {
  state = {
    endDate: null,
    focusedInput: null,
    startDate: null,
  };

  /**
   * Call `props.onChange` with the new dates from state.
   */
  componentDidUpdate = (prevProps, prevState) => {
    const prevDates = pickDatesFromState(prevState);
    const dates = pickDatesFromState(this.state);
    const { name, onChange } = this.props;
    !isEqual(prevDates, dates) && onChange(name, dates);
  };

  /**
   * Persist the focused input identifier in component state.
   */
  handleFocusChange = focusedInput => this.setState({ focusedInput });

  /**
   * Persist the date values in component state.
   */
  handleInputControllerChange = (name, value) => {
    this.setState(value);
  };

  render() {
    const {
      displayFormat,
      endDatePlaceholderText,
      error,
      getIsDayBlocked,
      isValid,
      name,
      startDatePlaceholderText,
    } = this.props;
    const { endDate, focusedInput, startDate } = this.state;
    return (
      <InputController
        error={error}
        inputOnChangeFunctionName="onDatesChange"
        isFocused={!!focusedInput}
        isValid={isValid}
        name={name}
        onChange={this.handleInputControllerChange}
      >
        <DateRangePicker
          // Consumer defined props.
          displayFormat={displayFormat}
          endDatePlaceholderText={endDatePlaceholderText}
          isDayBlocked={getIsDayBlocked}
          startDatePlaceholderText={startDatePlaceholderText}
          // Controlled props
          endDate={endDate}
          focusedInput={focusedInput}
          // NOTE onDatesChange is required by DateRangePicker but is set in `InputController`
          onDatesChange={Function.prototype}
          onFocusChange={this.handleFocusChange}
          startDate={startDate}
          // Static required props.
          endDateId={uniqueId('end_date_id_')}
          startDateId={uniqueId('start_date_id_')}
          // Static custom appearance props.
          customArrowIcon={<Icon name="arrow right" />}
          customInputIcon={<Icon name="calendar outline" />}
          daySize={52}
          hideKeyboardShortcutsPanel
          navNext={<Icon name="arrow right" />}
          navPrev={<Icon name="arrow left" />}
        />
      </InputController>
    );
  }
}

Component.displayName = 'DateRangePicker';

Component.defaultProps = {
  displayFormat: 'DD/MM/YYYY',
  endDatePlaceholderText: '',
  error: false,
  getIsDayBlocked: Function.prototype,
  isValid: false,
  name: '',
  onChange: Function.prototype,
  startDatePlaceholderText: '',
};

Component.propTypes = {
  /** The format for inputting and displaying dates as text. */
  displayFormat: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /** The visible placeholder text for the end date input. */
  endDatePlaceholderText: PropTypes.string,
  /** Is the date range picker in an error state. A string is displayed as an error message. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /**
   * A function called for each day to be displayed. Returning true blocks that day in the date range picker.
   * @param   {Moment}  day - The day to test.
   * @returns {Boolean}     - Is the day blocked.
   */
  getIsDayBlocked: PropTypes.func,
  /** Is the date range picker in a valid state. */
  isValid: PropTypes.bool,
  /** The name for the date range picker. */
  name: PropTypes.string,
  /**
   * A function called when the date range picker value changes. Dates are wrapped as [Moment objects](https://momentjs.com/docs/#/parsing/).
   * @param {String} name   - `this.props.name`
   * @param {Object} value
   * @param {Moment} value.endDate
   * @param {Moment} value.startDate
   */
  onChange: PropTypes.func,
  /** The visible placeholder text for the start date input. */
  startDatePlaceholderText: PropTypes.string,
};
