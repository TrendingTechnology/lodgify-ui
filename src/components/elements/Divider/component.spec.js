import React from 'react';
import { shallow } from 'enzyme';
import { Divider as SemanticDivider } from 'semantic-ui-react';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as Divider } from './component';

const getDivider = props => shallow(<Divider {...props} />);

describe('<Divider />', () => {
  it('should render a single Semantic UI `Divider` component', () => {
    const wrapper = getDivider();
    const actual = wrapper.find(SemanticDivider).length;
    expect(actual).toBe(1);
  });

  describe('the Semantic UI `Divider` component', () => {
    it('should get the right props', () => {
      const wrapper = getDivider();
      const actual = wrapper.find('Divider').props();
      expect(actual).toEqual(
        expect.objectContaining({
          hidden: true,
        })
      );
    });

    it('should get `props.hidden` as false if the parent `props.hasLine` is true', () => {
      const wrapper = getDivider({ hasLine: true });
      const actual = wrapper.find('Divider').props();
      expect(actual).toEqual(
        expect.objectContaining({
          hidden: false,
        })
      );
    });
  });

  it('should have `displayName` Divider', () => {
    expectComponentToHaveDisplayName(Divider, 'Divider');
  });
});
