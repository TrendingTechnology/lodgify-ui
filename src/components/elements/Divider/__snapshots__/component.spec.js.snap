// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> by default should render the right structure 1`] = `
<Divider
  className=""
  hidden={true}
  section={false}
/>
`;

exports[`<Divider /> if \`props.hasLine === true\` should render the right structure 1`] = `
<Divider
  className=""
  hidden={false}
  section={false}
/>
`;

exports[`<Divider /> if \`props.size === "large"\` should render the right structure 1`] = `
<Divider
  className=""
  hidden={true}
  section={true}
/>
`;
